src/protocol/http/src/main/java/org/apache/jmeter/protocol/http/proxy/ProxyControl.java

```diff
@@ -355,6 +399,54 @@ public void setHTTPSampleNamingMode(int httpNamingMode) {
         setProperty(new IntegerProperty(HTTP_SAMPLER_NAMING_MODE, httpNamingMode));
     }
 
+    public void setHttpSampleNumberingMode(int httpSampleNumberingMode) {
+        setProperty(new IntegerProperty(HTTP_SAMPLER_NUMBERING_MODE, httpSampleNumberingMode));
+        if (httpSampleNumberingMode == 0) {
+            // choose prefix numbering
+            JMeterUtils.setProperty("proxy.number.mode", "prefix"); // $NON-NLS-1$
+            JMeterUtils.setProperty("proxy.number.requests", "true"); // $NON-NLS-1$
+        }
+        if (httpSampleNumberingMode == 1) {
+            // choose suffix numbering
+            JMeterUtils.setProperty("proxy.number.mode", "suffix"); // $NON-NLS-1$
+            JMeterUtils.setProperty("proxy.number.requests", "true"); // $NON-NLS-1$
+        }
+        if (httpSampleNumberingMode == 2) {
+            // choose no numbering
+            JMeterUtils.setProperty("proxy.number.requests", "false"); // $NON-NLS-1$
+        }
+    }
+
+    public void setHttpSamplerNumberingStartValue(int httpSamplerNumberingStartValue) {
+        iNumberingStartValue=  httpSamplerNumberingStartValue;
+    }
+
+    public void setHttpSamplerNumberingStartForSamplerCreator(int httpSamplerNumberingStartValue) {
+        iNumberingStartValue =  httpSamplerNumberingStartValue;
+
+        SamplerCreator samplerCreator = samplerCreatorFactory.getDefaultSamplerCreator();
+        if (samplerCreator instanceof DefaultSamplerCreator) {
+            DefaultSamplerCreator defaultSamplerCreator = (DefaultSamplerCreator) samplerCreator;
+            defaultSamplerCreator.setRequestNumber(httpSamplerNumberingStartValue);
```

Comment:
setRequestNumber should be a method of SamplerCreator interface with a default implementation in AbstractSamplerCreator. So you would not have to have to check instanceof